processing_handler.go

```diff
@@ -303,8 +318,10 @@ func handleProcessing(reqID string, rw http.ResponseWriter, r *http.Request) {
 
 	if err == nil {
 		defer originData.Close()
-	} else if nmErr, ok := err.(*imagedata.ErrorNotModified); ok && config.ETagEnabled {
-		rw.Header().Set("ETag", etagHandler.GenerateExpectedETag())
+	} else if nmErr, ok := err.(*imagedata.ErrorNotModified); ok {
+		if config.ETagEnabled {
```

Comment:
We should also check `len(etagHandler.ImageEtagExpected()) != 0`.
We didn't need this check before because we couldn't reach this point unless we set `If-None-Match`, but now we can.

Reply:
done!